### PR TITLE
Markup for phase editing in groups

### LINKED ISSUE #1234: Process admin page / Modify info about phases
- In the phase list, show more info about the phases in the process initial page
- Direct links to tools created
- Direct link to create a new tool instance
- Remove tools tabs

---

- [x] Design
- [ ] Implementation

### PR DESCRIPTION
**Pull request by furilo**
Closes #1234 #1236 #1141 #1237 #1239 #1235 #1238

### What does this PR do?

- [x] Markup for phase editing
- [x] Implementation

### How should this be manually tested?

- Markup: Browse to http://madrid.gobierto.dev/sandbox/admin_part_group

### Alvaro's revision

- [x] Add to select News and Documents
- [x] Change Meetings with Agenda
- [x] New issue Link phases with pages (select2)
- [x] New tab phases
- [x] Change “Titulo de una fase”, “Descripción de una fase”
- [x] Not include “CTA box” and “Phases Map” when is group, also the dates of the phases.
- [x] New issue -> Highlight open phases

### CODE REVIEW

File: app/assets/stylesheets/admin/application.scss [hunk 1]
```diff
@@ -23,7 +23,7 @@
 @import "simplemde.min";
 @import "tipsy";
 @import "jqtree";
-@import "codemirror";
+// @import "codemirror";
```
Comment (ferblape) [inline app/assets/stylesheets/admin/application.scss]: Is this correct?

File: app/controllers/gobierto_admin/gobierto_participation/processes/process_stage_pages_controller.rb [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+                                                                                               process_stage_id: @process_stage_page_form.process_stage_page.process_stage), notice: t(".success")
+          else
+            render :update
+          end
+        end
+
+        private
+
+        def find_process_stage
+          ::GobiertoParticipation::ProcessStage.find(params[:process_stage_id])
```
Comment (ferblape) [inline app/controllers/gobierto_admin/gobierto_participation/processes/process_stage_pages_controller.rb]: Scope: `current_process.stages.find...`

File: app/controllers/gobierto_admin/gobierto_participation/processes/process_stage_pages_controller.rb [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        end
+
+        def find_process_stage_page
+          ::GobiertoParticipation::ProcessStagePage.find(params[:id])
```
Comment (ferblape) [inline app/controllers/gobierto_admin/gobierto_participation/processes/process_stage_pages_controller.rb]: Scope: `@process_stage.process_stage_page...`

File: app/controllers/gobierto_admin/gobierto_participation/processes/process_stages_controller.rb [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+            description_translations: [*I18n.available_locales]
+          )
+        end
+
+        def ignored_process_stage_attributes
+          %w(active position created_at updated_at)
+        end
+
+        def find_process_stage
+          ::GobiertoParticipation::ProcessStage.find(params[:id])
```
Comment (ferblape) [inline app/controllers/gobierto_admin/gobierto_participation/processes/process_stages_controller.rb]: Scopes...

File: app/forms/gobierto_admin/gobierto_participation/process_form.rb [hunk 5]
```diff
@@ -143,27 +134,22 @@ def calculate_has_duration(options)
...
-          title_translations: attributes['title_translations'],
-          description_translations: attributes['description_translations'],
-          cta_text_translations: attributes['cta_text_translations'],
-          active: attributes['active'],
-          starts: attributes['starts'],
-          ends: attributes['ends']
+      def build_information_stage
+        process.stages.build(
+          process: process,
+          title_translations: { 'en' => 'Information', 'es' => 'Información' },
```
Comment (ferblape) [inline app/forms/gobierto_admin/gobierto_participation/process_form.rb]: Could we add Catalan?

File: app/forms/gobierto_admin/gobierto_participation/process_stage_form.rb [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+      def process_stage
+        @process_stage ||= process_stage_class.find_by(id: id).presence || build_process_stage
+      end
+
+      def process_id
+        @process_id ||= process_stage.process_id
+      end
+
+      def process
+        @process ||= GobiertoParticipation::Process.find_by(id: process_id)
```
Comment (ferblape) [inline app/forms/gobierto_admin/gobierto_participation/process_stage_form.rb]: `current_site.processes`

File: app/helpers/gobierto_admin/gobierto_participation/process_stages_helper.rb [hunk 7]
```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+module GobiertoAdmin
+  module GobiertoParticipation
+    module ProcessStagesHelper
+      def admin_stage_url(stage)
```
Comment (ferblape) [inline app/helpers/gobierto_admin/gobierto_participation/process_stages_helper.rb]: The method returns paths, it should be named `admin_stage_path`?

**Review by ferblape — CHANGES_REQUESTED** [review]
Good job! 

Just review my comments. Most of them are scopes in the queries (we should avoid this error from now on ;)

And a couple of things:

- When editing the phases of the process, in the breadcrumb the title of the process is not visible

- When editing a stage in the breadcrumb says CMS

- In `Process` model:
  - `next_stage` should use the new ordering, not the date
  - same with `showcase_stage`
  - in `current_stage`: remove the comment
  - `self.open` gets all the processes of all the sites. That method is quite dangerous. Is it used somewhere?

- The URL of the Information and Results page should keep the user in the process. Now, it takes the user to the CMS section, and it should be the show action inside the process.

File: app/assets/stylesheets/admin/application.scss (hunk 1 above)
Comment (apradillap) [reply app/assets/stylesheets/admin/application.scss]: Alvaro, estuvo probando "Remove the Codemirror CSS temporarily to see how it feels"
https://github.com/PopulateTools/gobierto/pull/1248/commits/7ae4ea97c093b7572884a79d4422a4c03f09168d @furilo, what do we do?

**Comment by furilo** [comment]
>
> Alvaro, estuvo probando "Remove the Codemirror CSS temporarily to see how
> it feels"
> 7ae4ea9
> <https://github.com/PopulateTools/gobierto/commit/7ae4ea97c093b7572884a79d4422a4c03f09168d>
> @furilo <https://github.com/furilo>, what do we do?
>

Don’t use it, we can remove the commented reference


-- 
http://twitter.com/furilo
http://www.furilo.com

**Comment by apradillap** [comment]
@ferblape I've corrected all the "requested changes" except `next_stage` and `showcase_stage` in `Process` model that are returning stages 😄 
Highlights: I deleted `self.open` dangerous method, I'll check the scopes in future PRs, thanks!!

**Comment by ferblape** [comment]
Thanks @apradillap just two more details I have just noticed:

screen shot 2017-12-29 at 07 20 27

1. "More information" button should only be visible if there's a phase with the Information tool published. Otherwise, the link is broken and raises this error: https://rollbar.com/Populate/gobierto/items/692/

2. The markup in the stages navigator for the first stage is wrong. In the example above the active stage is the second one, but the first stage is bold too, and the user can't distinguish which is the active stage.

Feel free to release after fixing these two issues 🚀

**Comment by ferblape** [comment]
Also @apradillap could you reference in the PR all the issues it closes? 1238, 1235....

**Pull request merged**